client-frameworks-support/client-support-angular/projects/client-support-angular/README.md

```diff
@@ -1,24 +1,112 @@
+LuigiContextService is an abstract class, its implementation is in LuigiContextServiceImpl class.  
+If you need to change/extend the implementation, you can easily create your a new class extending LuigiContextServiceImpl:
 
-After building your library with `ng build client-support-angular`, go to the dist folder `cd dist/client-support-angular` and run `npm publish`.
+```javascript
+export class YourContextService extends  LuigiContextServiceImpl {
+    ....    
+}
 
-## Running unit tests
+```
+Inside your module redefine the provider:
+ ```javascript
+providers: [
+    {
+        provide: LuigiContextService,
+        useClass: YourContextService
+    }
+]
+ ```
+    
+### LuigiAutoRoutingService
+This service you cannot directly be used, but it will provide useful features how to synchronize your angular application with Luigi navigation.  
+It can happen that in your microfrontend, user can navigate through different components/pages.  
+With this feature we provide an easy way how to synchronize angular route with Luigi navigation; in angular route configuration, you can now add in data these attributes:
 
-Run `ng test client-support-angular` to execute the unit tests via [Karma](https://karma-runner.github.io).
+ ```javascript
+{path: 'luigi-client-support-preload',component: Sample1Component,data: { fromVirtualTreeRoot: true }}
+{path: 'luigi-client-support-preload',component: Sample2Component,data: { luigiRoute: '/home/sample2' }}
+ ```
 
-## Further help
+with `data: { fromVirtualTreeRoot: true }`, once we load Sample1Component, we will call Luigi Client:
+ ```javascript
+  luigiClient.linkManager().fromVirtualTreeRoot().withoutSync().navigate({route url});
+ ```
+with `data: { luigiRoute: '/home/sample2'' }`, uses luigiClient API in this way:
+ ```javascript
+  luigiClient.linkManager().withoutSync().navigate(data.luigiRoute);
+ ```
+More information about linkManager can be found [here](https://docs.luigi-project.io/docs/luigi-client-api/?section=linkmanager).
 
-To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI Overview and Command Reference](https://angular.io/cli) page.
+
+## LuigiRouteStrategy
+To use LuigiAutoRoutingService, this library defines a new RouteReuseStrategy named LuigiRouteStrategy.  
+If you need to define your own RouteReuseStrategy, you can extend LuigiRouteStrategy by overriding it next way:
+
+ ```javascript
+export class YouRouteStrategy extends LuigiRouteStrategy {
+
+    retrieve(route: ActivatedRouteSnapshot): DetachedRouteHandle {
+        super.retrieve(route);
+        // Your code
+    }
+
+}
+ ```
+and define the provider:
+ ```javascript
+ {
+      provide: RouteReuseStrategy,
+      useClass: YouRouteStrategy
+ }
+ ```
+
+We also provide an example of how to extend LuigiRouteStrategy in class LuigiReuseRouteStrategy.  
+In this class we added the possibility to "reuse" a component and not init every time you load it (it could be useful to keep component state.)  
```

Comment:
```suggestion
In this class, we added the possibility to "reuse" a component without re-initializing it on load every time (it could be useful to keep the component state.)  
```